### PR TITLE
show only logs with a severity level of ERROR or higher in the stderr.

### LINKED ISSUE #4049: metrics-apiserver json logs
> I'm very sorry @JorTurFer - I was looking at logs from the `keda-operator-metrics-apiserver` deployment. It seems that only has a level setting as opposed to the operator having level, format, and timeEncoding.
> 
> Our logs for `keda-operator` are correctly being emitted as JSON 🤦

That said, I'd like to have the metrics-apiserver log in json format as well, primarily because all info logs are emitted to stderr and those show up as error logs in our logging system. Could this functionality please be added?

_Originally posted by @naseemkullah in https://github.com/kedacore/keda/issues/3655#issuecomment-1367852952_

### PR DESCRIPTION
**Pull request by Adarsh-verma-14**
Signed-off-by: Adarsh-verma-14 [t_adarsh.verma@india.nec.com](mailto:t_adarsh.verma@india.nec.com)

In the logs of keda-metrics-apiserver, show only log messages  with a severity level of ERROR or higher in the stderr and Log messages with severity levels below ERROR will be included in the stdout stream. 

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Fixes #4049

### CODE REVIEW

**Review by JorTurFer — COMMENTED** [review]
As this is to solve an issue, could you update the changelog as well with the reference to the issue?

We need to update this also in [helm chart](https://github.com/kedacore/charts/blob/main/keda/templates/metrics-server/deployment.yaml#L115), could you open another PR there?

File: config/metrics-server/deployment.yaml [hunk 1]
```diff
@@ -57,7 +57,7 @@ spec:
           args:
           - /usr/local/bin/keda-adapter
           - --secure-port=6443
-          - --logtostderr=true
+          - --stderrthreshold=ERROR
```
[the remark this answers is not in the document]
Comment (zroubalik) [reply config/metrics-server/deployment.yaml:60]: Agree with @JorTurFer here

[the remark this answers is not in the document]
Comment (Adarsh-verma-14) [reply config/metrics-server/deployment.yaml:60]: Absolutely agree with @JorTurFer, if I use both arg then the output format of the logs will remain in the stderr , as the --logtostderr=true arg explicitly directs the log output to the stderr.
However, the --stderrthreshold=ERROR arg will filter the logs based on their severity levels. Only logs with a severity level of ERROR or higher will be displayed in the stderr. Logs with severity levels below ERROR will not be shown in the stderr even though they are directed to stderr.

**Comment by Adarsh-verma-14** [comment]
sorry something is wrong from my side I create again PR regarding this.

**Comment by Adarsh-verma-14** [comment]
I have closed this PR and also raised a new PR(#4944 ) for the same issue .

**Pull request closed without merging**